### PR TITLE
chore: Test against ARIA practices

### PR DESCRIPTION
**Pull request by WilcoFiers**
One of the more common reasons for reported false positives comes from axe-core failing something recommended in the ARIA practices. This PR adds a test script will have axe-core running against the ARIA practices.

- [x] Set up basic test script
- [x] Open issues for problems discovered in the test:apg
- [x] Lock the APG dependency to a fix commit for stable testing
- [x] Set up CI, including a nightly test where we're running against the latest on #main

Closes issue: #2918

### CODE REVIEW

**Comment by WilcoFiers** [comment]
PR is blocked until the aria-practices removes husky from its postinstall script. I opened a PR for that: https://github.com/w3c/aria-practices/pull/2033

File: .circleci/config.yml [hunk 1]
```diff
@@ -354,6 +379,9 @@ workflows:
                 - develop
     jobs:
       - dependencies_unix
-      - test_nightly:
+      - test_nightly_chrome:
```
Comment (stephenmathieson) [inline .circleci/config.yml:382]: How will we know if/when this fails? Is there someone responsible for checking CI logs daily?

File: test/aria-practices/apg.spec.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+const express = require('express');
+const chromedriver = require('chromedriver');
+const AxeBuilder = require('@axe-core/webdriverjs');
+const testListen = require('test-listen');
+const { getWebdriver, connectToChromeDriver } = require('./run-server');
+const { assert } = require('chai');
+const globby = require('globby');
+
+describe('aria-practices', function () {
+  const apgPath = path.resolve(__dirname, '../../node_modules/aria-practices/');
```
Comment (stephenmathieson) [inline test/aria-practices/apg.spec.js:13]:
```suggestion
  const apgPath = require.resolve('aria-practices')
```

File: test/aria-practices/apg.spec.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      
+      const builder = new AxeBuilder(driver, axeSource);
+      builder.disableRules([
+        ...disabledRules['*'],
+        ...(disabledRules[filePath] || []),
+      ]);
+      
+      const { violations } = await builder.analyze();
+      const issues = violations.map(({ id, nodes }) => ({ id, issues: nodes.length }))
+      assert.deepEqual(issues, []);
```
Comment (stephenmathieson) [inline test/aria-practices/apg.spec.js:82]:
```suggestion
      assert.lengthOf(issues, 0);
```

File: .circleci/config.yml (hunk 1 above)
Comment (WilcoFiers) [reply .circleci/config.yml:382]: Yeah, those get kicked out to our #attest-build channel on Slack.

File: test/aria-practices/apg.spec.js (hunk 2 above)
Comment (WilcoFiers) [reply test/aria-practices/apg.spec.js:13]: Gd point. I originally started with those in another dir, but this is better.

Comment (WilcoFiers) [reply test/aria-practices/apg.spec.js:13]: Never mind. I remember why. aria-practices doesn't have a package.json. Node complains about that.

File: .circleci/config.yml [hunk 4]
```diff
@@ -5,7 +5,7 @@ defaults: &defaults
 
 unix_box: &unix_box
   docker:
-    - image: circleci/node:10-browsers
+    - image: circleci/node:16-browsers
```
Comment (WilcoFiers) [inline .circleci/config.yml:8]: So that we get access to Chrome 95

File: test/aria-practices/apg.spec.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  this.retries(3);
+
+  before(async () => {
+    const axePath = require.resolve('../../axe.js');
+    axeSource = fs.readFileSync(axePath, 'utf8');
+    chromedriver.start([`--port=${port}`]);
+    await new Promise(r => setTimeout(r, 500));
+    await connectToChromeDriver(port);
+
+    const app = express();
```
Comment (straker) [inline test/aria-practices/apg.spec.js:29]: Are any of the test files using absolute paths? If not I think our `npm run start` command should server node_modules directly as well, so we might not need express for this.

File: package.json [hunk 6]
```diff
@@ -106,16 +107,18 @@
     "fmt": "prettier --write .*.json *.{json,md,js} **/*.ts './{.circleci,.github,build,doc,lib,locales,test,typings}/**/*.{json,md,js,ts,html}'"
   },
   "devDependencies": {
+    "@axe-core/webdriverjs": "^4.2.2",
```
[the remark this answers is not in the document]
Comment (WilcoFiers) [reply package.json:110]: Don't think so. It's a dev dependency.

**Review by straker — CHANGES_REQUESTED** [review]

**Review by straker — APPROVED** [review]

**Pull request merged**